src/common/strtol.h

```diff
@@ -48,4 +48,20 @@ Target strict_si_cast(const char *str, std::string *err) {
 template <>
 uint64_t strict_si_cast(const char *str, std::string *err);
 
+template<typename T, const unsigned base = 10, const unsigned width = 1>
```

Comment:
1. Yes, it does. We 100% avoid memory reference, otherwise it's up to compiler and context.
2. On my taste the interface is well balance between performance, readability, easy of implementation and use. If you have better idea, we can discuss.